iceoryx2-cli/iox2-config/src/commands.rs

```diff
@@ -0,0 +1,127 @@
+pub fn print_system_configuration() {
+    const HEADER_COLOR: &str = "\x1b[4;92m";
+    const VALUE_COLOR: &str = "\x1b[0;94m";
+    const DISABLED_VALUE_COLOR: &str = "\x1b[0;90m";
+    const ENTRY_COLOR: &str = "\x1b[0;37m";
+    const DISABLED_ENTRY_COLOR: &str = "\x1b[0;90m";
+    const COLOR_RESET: &str = "\x1b[0m";
+
+    println!("{}posix system configuration{}", HEADER_COLOR, COLOR_RESET);
+    println!();
+    println!(" {}system info{}", HEADER_COLOR, COLOR_RESET);
+    for i in all::<SystemInfo>().collect::<Vec<_>>() {
+        println!(
+            "  {ENTRY_COLOR}{:<50}{COLOR_RESET} {VALUE_COLOR}{}{COLOR_RESET}",
+            format!("{:?}", i),
+            i.value(),
+        );
+    }
+
+    println!();
+    println!(" {}limits{}", HEADER_COLOR, COLOR_RESET);
+    for i in all::<Limit>().collect::<Vec<_>>() {
+        let limit = i.value();
+        let limit = if limit == 0 {
+            "[ unlimited ]".to_string()
+        } else {
+            limit.to_string()
+        };
+        println!(
+            "  {ENTRY_COLOR}{:<50}{COLOR_RESET} {VALUE_COLOR}{}{COLOR_RESET}",
+            format!("{:?}", i),
+            limit,
+        );
+    }
+
+    println!();
+    println!(" {}options{}", HEADER_COLOR, COLOR_RESET);
+    for i in all::<SysOption>().collect::<Vec<_>>() {
+        if i.is_available() {
+            println!(
+                "  {ENTRY_COLOR}{:<50}{COLOR_RESET} {VALUE_COLOR}{}{COLOR_RESET}",
+                format!("{:?}", i),
+                i.is_available(),
+            );
+        } else {
+            println!(
+                "  {DISABLED_ENTRY_COLOR}{:<50}{COLOR_RESET} {DISABLED_VALUE_COLOR}{}{COLOR_RESET}",
+                format!("{:?}", i),
+                i.is_available(),
+            );
+        }
+    }
+
+    println!();
+    println!(" {}features{}", HEADER_COLOR, COLOR_RESET);
+    for i in all::<Feature>().collect::<Vec<_>>() {
+        if i.is_available() {
+            println!(
+                "  {ENTRY_COLOR}{:<50}{COLOR_RESET} {VALUE_COLOR}{}{COLOR_RESET}",
+                format!("{:?}", i),
+                i.is_available(),
+            );
+        } else {
+            println!(
+                "  {DISABLED_ENTRY_COLOR}{:<50}{COLOR_RESET} {DISABLED_VALUE_COLOR}{}{COLOR_RESET}",
+                format!("{:?}", i),
+                i.is_available(),
+            );
+        }
+    }
+
+    println!();
+    println!(" {}process resource limits{}", HEADER_COLOR, COLOR_RESET);
+    for i in all::<ProcessResourceLimit>().collect::<Vec<_>>() {
+        println!(
+            "  {ENTRY_COLOR}{:<43}{COLOR_RESET} soft:  {VALUE_COLOR}{:<24}{COLOR_RESET} hard:  {VALUE_COLOR}{}{COLOR_RESET}",
+            format!("{:?}", i),
+            i.soft_limit(),
+            i.hard_limit()
+        );
+    }
+}
+
+pub fn show() -> Result<()> {
+    print_system_configuration();
+
+    Ok(())
+}
+
+pub fn generate() -> Result<()> {
```

Comment:
In `iceoryx2/src/config.rs` there is the `DEFAULT_CONFIG_FILE` defined. This needs to be adjusted so that it looks up all the XDG paths - maybe it reads them from an env var like @elBoberido suggested.

Another approach is, that the config path goes in as additional command line parameter and is then parsed and provided to the node like this here:

```rust
let config = Config::from_file(path_to_config_file)?;
let node = NodeBuilder::new()
     .config(&config)
     .create::<ipc::Service>()?;
```

Reply:
I think it looks up all the XDG paths but when there is no configuration file what happens ? A default configuration is used and a configuration file is generated OR an error is generated and ask the user to generate a default configuration file with the CLI tool ?

Reply:
@brosier01 If there is no configuration found in the location it is told to look, it uses the in-built default configuration and a debug log is printed.

Reply:
I think we have to differentiate between two scenarios
- a location was provided and the file was not there -> an error message is printed
- no location was provided and there is no file at the default locations -> a debug message is printed

Reply:
@elBoberido The current behaviour is that the built-in default is used when a config is not found at the default location. Did you want to change this ? To minimize scope creep, maybe better left for a separate PR.

Reply:
@elfenpiff I think `DEFAULT_CONFIG_FILE` should not be used. If I understand that code correctly, it looks at a relative location to where the application was started.

It should always be deterministic and always use a fixed reference, e.g. `$HOME` or `/etc` on Linux.

Ideally, it would just traverse the entries in `XDG_DATA_DIRS` from the most specific location to the mots generic one. Unfortunately, reading an environment variable allocates, which means we would to refactor that later on when do the certification. It could be hidden behind a feature flag though and if disabled, we would provide a fixed list of predefined locations by ourselves.

Reply:
As I understand it, the work to be done is as follows: 
- DEFAULT_CONFIG_FILE must be adjusted so that it searches all XDG paths. (no need to provide a specific location)
- If there is no configuration file: a debug printout is used and the built-in configuration is used.

Reply:
@brosier01 Yes this would be good. If this is more complicated than it seems, let us know.

Reply:
Generated config is in use now, nice.